src/datepicker/datepicker.js

```diff
@@ -576,16 +577,29 @@ function ($compile, $parse, $document, $position, dateFilter, dateParser, datepi
 
       scope.$watch('isOpen', function(value) {
         if (value) {
-          scope.$broadcast('datepicker.focus');
-          scope.position = appendToBody ? $position.offset(element) : $position.position(element);
-          scope.position.top = scope.position.top + element.prop('offsetHeight');
+          if (!scope.ngDisabled) {
+            scope.$broadcast('datepicker.focus');
+            scope.position = appendToBody ? $position.offset(element) : $position.position(element);
+            scope.position.top = scope.position.top + element.prop('offsetHeight');
 
-          $document.bind('click', documentClickBind);
+            $document.bind('click', documentClickBind);
+          }
+          else {
+            scope.isOpen = false;
+          }
         } else {
           $document.unbind('click', documentClickBind);
         }
       });
 
+      scope.$watch('ngDisabled', function(value) {
```

Comment:
Should be able to do `scope.$parent.$watch(attrs.ngDisabled`. Need to use parent because this is an isolate scope.